src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3WebtransportManager.cs

```
@@ -0,0 +1,123 @@
+        {
+            _sessions = new ConcurrentDictionary<long, Http3WebtransportSession>();
+            _connection = connection;
+        }
+
+        public bool AddSession(QuicStream connectStream, Http3WebtransportSession webtransportSession)
+        {
+            bool ans = _sessions.TryAdd(connectStream.Id, webtransportSession);
+            return ans;
+        }
+
+        public bool FindSession(long streamId, out Http3WebtransportSession? session)
+        {
+            return _sessions.TryGetValue(streamId, out session);
+        }
+
+        public void AcceptServerStream(QuicStream stream, long sessionId)
+        {
+            Http3WebtransportSession? session;
+            _sessions.TryGetValue(sessionId, out session);
+            // if no session with that id exists throw exception
+            // https://datatracker.ietf.org/doc/html/draft-ietf-webtrans-http3#section-4
+            if (session == null)
+            {
+                stream.Abort(QuicAbortDirection.Both, (long)Http3ErrorCode.WebtransportBufferedStreamRejected);
+                return;
+            }
+            session.AcceptServerStream(stream);
+        }
+
+        public async ValueTask<QuicStream?> CreateClientStream(QuicStreamType type, long sessionId)
+        {
+            QuicStream clientWTStream;
+            try
+            {
+                clientWTStream = await _connection.OpenOutboundStreamAsync(type).ConfigureAwait(false);
+                if (type == QuicStreamType.Unidirectional)
+                    await clientWTStream.WriteAsync(BuildWebtransportStreamClientFrame(type, sessionId), default).ConfigureAwait(false);
+                else
+                    await clientWTStream.WriteAsync(BuildWebtransportStreamClientFrame(type, sessionId), default).ConfigureAwait(false);
+
+                return clientWTStream;
+            }
+            catch (Exception)
+            {
+                return null;
+            }
```

You shouldn't silently swallow exceptions unless you have a very good reason to do so. Especially *all* exceptions like here